File: generator/registry/playwright-python/files/pages/base_page.py
"""BasePage: navigation and helpers shared by every page object."""
from playwright.sync_api import Page


class BasePage:
    def __init__(self, page: Page, base_url: str) -> None:
        self.page = page
        self.base_url = base_url

    def goto(self, path: str = "/") -> None:
        self.page.goto(f"{self.base_url}{path}")

    def title(self) -> str:
        return self.page.title()

File: packages/generator/registry/cypress-js/files/cypress/pages/BasePage.js
/**
 * BasePage holds behaviour shared by every page object: navigation and
 * common helpers. Concrete pages extend this and expose intent-revealing
 * methods rather than raw selectors.
 */
class BasePage {
  visit(path = '/') {
    cy.visit(path);
    return this;
  }

  getByTestId(id) {
    return cy.get(`[data-testid="${id}"]`);
  }

  clickByText(text) {
    cy.contains(text).click();
    return this;
  }
}

module.exports = BasePage;
